src/docs/arch-rts.dox

```diff
@@ -1,113 +1,76 @@
-following order.
-1. Check smallest insert lists on the page
-2. Traverse through all the on-disk keys
-    a. Check update list
-    b. Check insert list
-    c. Check the on-disk version if no stable update found in the update list.
-3. Traverse through the reconciled pages to abort any history store updates.
+Finally, RTS is triggered on startup, shutdown or when the application is initiated. It is worth
+noting that it cannot run if there is any transaction activity happening in WiredTiger.
 
-@section rts-abort-update How rollback to stable aborts in-memory updates
+@section rts-pages Internal and leaf pages
 
-Traverse through all the updates in the update list and abort them until a stable
-update is found.
+On internal pages, RTS can roll back unstable fast truncate operations.
+On leaf pages, RTS can roll back unstable updates with the following steps:
 
-@section rts-abort-on-disk-update How rollback to stable aborts on-disk update
+1. Checks the smallest insert lists on the page
+2. Traverses through all the on-disk keys
+    a. Checks the update list
+    b. Checks the insert list
+    c. Checks the on-disk version if no stable update is found in the update list.
+3. Traverses through the reconciled pages to abort any history store updates.
 
-If the start time pair is not stable try to find a valid update from the history
-store that is stable to replace the on-disk version otherwise remove the on-disk
-key. In case if the stop time pair if exists and if its not stable, restore the
-on-disk update again into the update list.
+@section rts-abort-update How RTS aborts in-memory updates
 
-To remove any existing update, rollback to stable adds globally visible tombstone to
-the key update list and this key will get removed later during the reconciliation.
+RTS traverses through all the updates defined in the update list and aborts every update until a
+stable is found. To remove any unstable update, RTS adds globally a visible tombstone to the key
+update list and this key gets removed later during reconciliation.
 
-Note: As of now, rollback to stable don't work on removing on-disk columnar updates.
+@section rts-abort-on-disk-update How RTS aborts on-disk update
 
-@section rts-hs-search Rollback to stable history store search
+If the start time pair is not stable, RTS searches the history store to find a stable update to
+replace the on-disk version. If the history store does not contain any stable update, the on-disk
+key is removed. If the stop time pair exists and it is not stable, the on-disk update is restored
+into the update list.
 
-Rollback to stable searches the history store to find a stable update to replace an
-unstable update in the data store. It searches the history store with the given data
-store key with a maximum timestamp and traverse back till a stable update found. If
-no valid update is found the data store key is removed.
+As of now, RTS does not remove on-disk columnar updates.
```

Comment:
We now support column store also similar to row store. RTS works for all storage types except LSM.

Reply:
Got it, fixed in https://github.com/wiredtiger/wiredtiger/pull/8292/commits/b1167da0aff5a914dcf1929dbf271d0a43a24121